tilequeue/wof.py

```diff
@@ -689,30 +799,38 @@ def fetch_toi():
         if wof_neighbourhoods_to_fetch:
             self.logger.info('Fetching %d raw neighbourhoods ...' %
                              len(wof_neighbourhoods_to_fetch))
-            # TODO this will also need to return back the
-            # neighbourhoods that we failed to sync to update diffs
-            raw_neighbourhoods = self.fetcher.fetch_raw_neighbourhoods(
-                wof_neighbourhoods_to_fetch)
+            raw_neighbourhoods, failures = (
+                self.fetcher.fetch_raw_neighbourhoods(
+                    wof_neighbourhoods_to_fetch))
             self.logger.info('Fetching %d raw neighbourhoods ... done' %
                              len(wof_neighbourhoods_to_fetch))
         else:
             self.logger.info('No raw neighbourhoods found to fetch')
             raw_neighbourhoods = ()
 
-        # TODO the logic will need to be updated here
-        # now that we have to manage "funky" data, and even prior
-        # there's a hole where it's possible for the raw data to be
-        # incomplete/invalid in some way. Those will be skipped from
-        # insertion in our model, but we'll also need to prune our
-        # diff list appropriately because it drives the generation of
-        # tiles to expire.
-        # It's also possible for some data to become invalid/funky
-        # that wasn't before, in which case we'll want to ensure that
-        # it's removed from our model
-        # A separate pass is planned anyway to make it more robust
-        # once we detect fetch failures, and I think it's best to
-        # handle all these cases at once, but just making a note in
-        # here now to serve as a reminder
+        # we should just remove any neighbourhoods from add/update lists
+        # also keep track of these ids to remove from the diffs too
+        failed_wof_ids = set()
+        for failure in failures:
+            failure_wof_id = failure.wof_id
+            # TODO logging funky failures is noisy
+            # maybe we just log the count of funky failures?
```

Comment:
I would agree - if we're expecting that we might get quite a lot of these in the future, to log them every hour is just going to bloat the logs. Does this mean that `NeighbourhoodFailure` needs a flag for funkiness?